pkgs/development/interpreters/bqn/cbqn/default.nix

```diff
@@ -16,6 +17,13 @@ let
     rev = "3df8ae563a626ff7ae0683643092f0c3bc2481e5";
     hash = "sha256:0rh9qp1bdm9aa77l0kn9n4jdy08gl6l7898lncskxiq9id6xvyb8";
   };
+  replxx-submodule = fetchFromGitHub {
```

Comment:
I think we should package this as a separate library because it is. This means of course that we need to patch CBQN's build system likely. Another tricky bit is of course that dzaima's replxx is the fork of the base replxx library.

Reply:
Hmm, it will take a longer time than I expected... I'd like to close this draft for now.

Reply:
Hm, I haven't had a look yet. Maybe it is unreasonably hard and we should just stick with vendoring in the submodule for now. I'm just a bit concerned that we never end up fixing it (which would be a shame since splitting it up improves correctness & speeds up compilation).

Reply:
Ok, then I converted this draft to a PR with tiny changes.
(But this request isn't so important now. I made a flake ^^)

Reply:
fwiw, my replxx fork will likely deviate more and more from base replxx over time - it already handles case-insensitive partial autocompletion differently (quite important for BQN) among other things, and things I may do in the future include overhauling input parsing (allowing ctrl+backspace & ctrl+delete among other things, which'd require changing the behavior of the replxx include header), and making it use CBQNs UTF-8 routines instead of it rolling its own (which [have their issues](https://github.com/AmokHuginnsson/replxx/issues/12)), which'd make it inseparable from CBQN.